plugins/chat/app/models/chat/channel.rb

```diff
@@ -164,6 +164,27 @@ def latest_not_deleted_message_id(anchor_message_id: nil)
       SQL
     end
 
+    def mark_all_threads_as_read(user: nil)
+      DB.exec(<<~SQL, channel_id: self.id)
+        UPDATE user_chat_thread_memberships
```

Comment:
Sorry I'm not familiar with the code here but when threading is enabled, do all users have `user_chat_thread_membership` records for all threads?

Reply:
No they don't only for threads they have replied to or manually tracked (before threading is enabled this is any message they replied to), so this can potentially be a lot or only a few. Should be pretty fast though in SQL only, when I did the migration to create `user_chat_thread_memberships` for old records even on dev it was fast.

Reply:
O so `user_chat_thread_memberships` records are already being created even when threading is disabled?

Reply:
Yes and threads as well. We've been doing that in the background for months so that when people turn threading on they automatically have existing threads (especially for reply chains where multiple users replied in a chain back to an original message).